lib/pbench/server/__init__.py

```diff
@@ -64,6 +65,109 @@ def tstos(ts: float = None) -> str:
     return dt.strftime("%Y-%m-%dT%H:%M:%S-%Z")
 
 
+def timestamp() -> str:
+    """Generate a time stamp string.
+
+    Returns:
+        The current timestamp formatted as a string of the following form:
+            <YYYY>-<MM>-<DD>T<hh>:<mm>:<ss>-<TZ>
+    """
+    return tstos(_time())
+
+
+def _get_valid_path(
+    env_name: str, dir_val: str, logger: Union[None, Logger]
+) -> Optional[Path]:
+    """Get the fully resolved path from the given path.
+
+    If a logger is given, will emit error logs when the directory is not
+    found or the resolved name is not a directory.
+
+    Args:
+        env_name : Legacy environment name to display in error messages
+        dir_val : A directory value to resolve to a real directory
+        logger : Logger to use when low-level error messages should be
+            emitted
+
+    Returns:
+        A Path directory object of the directory on disk, None otherwise.
+    """
+    try:
+        dir_path = Path(dir_val).resolve(strict=True)
+    except FileNotFoundError:
+        if logger:
+            logger.error(
+                "The {} directory, '{}', does not resolve to a real location",
+                env_name,
+                dir_val,
+            )
+        return None
+    else:
+        if not dir_path.is_dir():
+            if logger:
+                logger.error(
+                    "The {} directory, does not resolve to a directory ('{}')",
+                    env_name,
+                    dir_path,
+                )
+            return None
+    return dir_path
+
+
+def get_resolved_dir(
+    env_name: str, dir_val: str, logger: Union[None, Logger]
+) -> Optional[Path]:
+    """Get the resolved directory associated with the environment name.
+
+    Args:
+        env_name : Legacy environment name to display in error messages
+        dir_val : A directory value to resolve to a real directory
+        logger : Logger to use when low-level error messages should be
+            emitted
+
+    Returns:
+        A Path directory object, None if the directory value does not
+            resolve to a real directory.
+    """
+    dir_path = _get_valid_path(env_name, dir_val, logger)
+    if not dir_path:
+        return None
+
+    return dir_path
+
+
+def _get_valid_dir_option(
+    conf: ConfigParser, env_name: str, section: str, option: str
```

Comment:
Because the only use of `self` when it was part of the class was for accessing the config object.  It is not a strong tie to the configuration class and it let's us consider invoking the method as part of the `PbenchServerConfig` constructor.